src/com/hms/flexyosisoftconnector/DataPoster.java

```diff
@@ -32,134 +32,150 @@ public void run()
    {
       shouldRun = true;
 
+      long lastUpdateTimeMs = 0;
+
       while(shouldRun)
       {
+          long currentTimeMs = System.currentTimeMillis();
+
+          /*
+           *  The queue is more efficient if it send out after a large amount of data has been added
+           *  With this logic of waiting 6 seconds per check, the internal queues of each tag has time
+           *  to build up before we look at those internal to send out all the data in them.
+           */
+          int secondsBetweenSendData = 6;
+          int convertToMilliseconds = 1000;
+          int waitTime = secondsBetweenSendData*convertToMilliseconds;
+          if ((currentTimeMs - lastUpdateTimeMs) >= waitTime) {

```

Comment:
a comment about this if would be helpful. To explain what this is comparing

Reply:
Im not sure if this was intentionally moved to its own commit but the indentation changes seem like they belong in this commit "Implemented new data queue".  Having an "indentation commit is strange was the function change happened in this commit.